### DISCUSSION TITLE
Why Github markdowns aren't working properly?

### DISCUSSION THREAD

**Discussion by rohitcoder** [remarkjs/remark#1111] [Q&A] [answered]
Have a look at the below image, this is how GitHub is printing below example markdown, I want to achieve samething

<img width="859" alt="Screenshot 2023-02-05 at 1 18 14 AM" src="https://user-images.githubusercontent.com/17665703/216786927-c7ad36dd-6ac6-4ef6-8a48-2e60481f617a.png">

```
## Overview
[idno/known](https://packagist.org/packages/idno/known) is an A social publishing platform

Affected versions of this package are vulnerable to Information Exposure in the `getContent ` and `postContent ` functions in `Idno/Pages/Admin/Homepage.php`, which don't restrict access to Admins only. An authenticated user can expose information via insecure direct object reference (IDOR).
## Remediation
A fix was pushed into the `master` branch but not yet published.
## References
- [GitHub Commit](https://github.com/idno/known/pull/3101/commits/0b06c2fb05dcb9aa154ba078b560021fc07e7c03)
- [GitHub PR](https://github.com/idno/known/pull/3101)
- [Vulnerability Report](https://blog.jitendrapatro.me/multiple-vulnerabilities-in-idno-known-php-cms-software/#The-Vulnerability)
```

This my React Code
```javascript
import ReactMarkdown from 'react-markdown'
import remarkGfm from 'remark-gfm'

<ReactMarkdown remarkPlugins={[remarkGfm]}>
## Overview
[idno/known](https://packagist.org/packages/idno/known) is an A social publishing platform

Affected versions of this package are vulnerable to Information Exposure in the `getContent ` and `postContent ` functions in `Idno/Pages/Admin/Homepage.php`, which don't restrict access to Admins only. An authenticated user can expose information via insecure direct object reference (IDOR).
## Remediation
A fix was pushed into the `master` branch but not yet published.
## References
- [GitHub Commit](https://github.com/idno/known/pull/3101/commits/0b06c2fb05dcb9aa154ba078b560021fc07e7c03)
- [GitHub PR](https://github.com/idno/known/pull/3101)
- [Vulnerability Report](https://blog.jitendrapatro.me/multiple-vulnerabilities-in-idno-known-php-cms-software/#The-Vulnerability)
</ReactMarkdown>
```

And this is output (which is wrong)
I checked it's taking every thing as `<h2>` it should only take `<h2>` for "Overview" word
<img width="1063" alt="Screenshot 2023-02-05 at 1 20 42 AM" src="https://user-images.githubusercontent.com/17665703/216786990-6d631c92-970f-4932-b665-0e9c81732715.png">

**Comment by rohitcoder** [comment]
I tried to fix this, now this is only converting first line and rest are under ``<Pre><Code>``
<img width="946" alt="Screenshot 2023-02-05 at 1 43 38 AM" src="https://user-images.githubusercontent.com/17665703/216787853-42fdb349-a169-4e0e-b987-e3d6826a89a3.png">

Below is my new code
```javascript
    const markdown = `## Overview
    [idno/known](https://packagist.org/packages/idno/known) is an A social publishing platform

    Affected versions of this package are vulnerable to Information Exposure in the  direct object reference (IDOR).
    ## Remediation
    A fix was pushed into the  branch but not yet published.
    ## References
    - [GitHub Commit](https://github.com/idno/known/pull/3101/commits/0b06c2fb05dcb9aa154ba078b560021fc07e7c03)
    - [GitHub PR](https://github.com/idno/known/pull/3101)
    - [Vulnerability Report](https://blog.jitendrapatro.me/multiple-vulnerabilities-in-idno-known-php-cms-software/#The-Vulnerability)
    `;
<ReactMarkdown  children={markdown} />
```

**Reply by remcohaszing** [reply]
This is expected behaviour. Your markdown string starts with a level 2 heading, followed by a newline and an indentation of 4 spaces. In markdown, an indentation of 4 spaces is a code block.

To not treat the content as a code block, remove the indentation.

**Comment by remcohaszing** [comment, marked as answer]
This is expected behaviour. JSX strips newlines of text nodes. So you are telling remark to render this string, which is indeed a single markdown level 2 heading.

```js
"## Overview [idno/known](https://packagist.org/packages/idno/known) is an A social publishing platform Affected versions of this package are vulnerable to Information Exposure in the `getContent ` and `postContent ` functions in `Idno/Pages/Admin/Homepage.php`, which don't restrict access to Admins only. An authenticated user can expose information via insecure direct object reference (IDOR). ## Remediation A fix was pushed into the `master` branch but not yet published. ## References - [GitHub Commit](https://github.com/idno/known/pull/3101/commits/0b06c2fb05dcb9aa154ba078b560021fc07e7c03) - [GitHub PR](https://github.com/idno/known/pull/3101) - [Vulnerability Report](https://blog.jitendrapatro.me/multiple-vulnerabilities-in-idno-known-php-cms-software/#The-Vulnerability)"
```

For example, see the output in the [Babel playground](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=JYWwDg9gTgLgBAJQKYEMDGMCyKoGsAmEA7gHZwBmUEIcA5FKhgLQg4HEm0BQokscDVngDi5GpWp1BbJgHMx3LgB5k6LG0KkBSIbgAKAGwCus4CQDOAXgDeAbWkixAXQC-APi4BiT3ADyANyQof2AkIi5bYHwSCAB6XBjSJwAKAAsYGDBzAC5Y2LB0XBRTcxgAOmhZfMLipHNYqJj4xJIASjhgczgUMgBBOHMINGAUAzgwIwAjA07Us1lxgxQYcmgQLi5e8nIkDCR8OECoc2AICzgIcjgYOa6CtCLZJG6GQ6MDEiCUaeeYCDgAJIkVZQVgwU5kACiAA9IOYjK8zNdUs8AAZPGAAYTOMCQJHgqO6JAOqLhWJxeIJFCMJAwEK6SNRAOicT0tXqvXwIDMsQAEtQkAUnmUwKkwKiADRwIhzNCpOCETjwBilKDADDdNBoOpdP5wTnc85nAwATzK-rIKCMN0p6uW-zgRnMQTgaB6cCQsIgzo6wLWywhhxGvudaARz3wwAYGogkwAVrtlUgdgxac9kgCACK-BCtMpeHzIED7Ebgs6bCjAaHSlB3J0og5mPU2uCo1ilIKEyZQHpyuCTa1wGLwE1IeATaazfb57yIZNBPHa8xcJhwWzCYAwXlTODYkDcmApdKZHJ5Uw3KZlNDUBos5ocfLvAyxADMAEYAAxv2LX_eb-ofpMH4AGxoAATOQQEAKz4GgkwAJwoCgb5QQALJMKAfgA7AAHJMUHAR-H5gW-5BoNhSBYeRL6tCua4bluO56AgR4ZFkuSxOeqSXr-t5NAkD4TAYz7vl-tGrrYABq7yfD2kzADMMAmnOfCHmkbGnrE0wQLIZRxpueL4D2BQwFQZTFrEIDvOCYAGEgTD-DJXzyYpoTmEwZgeSyTACaQTCimATBoCA7mDOQMBEDgSCxJ4AAqKJMNJHzOQpm4mrRSixKoGDYHgmgkG4QA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=react&prettier=true&targets=&version=7.20.15&externalPlugins=&assumptions=%7B%7D).

To use multiline strings in JSX, use a JSX expression:

```js
import ReactMarkdown from 'react-markdown'
import remarkGfm from 'remark-gfm'

<ReactMarkdown remarkPlugins={[remarkGfm]}>{`
## Overview
[idno/known](https://packagist.org/packages/idno/known) is an A social publishing platform

Affected versions of this package are vulnerable to Information Exposure in the \`getContent \` and \`postContent \` functions in \`Idno/Pages/Admin/Homepage.php\`, which don't restrict access to Admins only. An authenticated user can expose information via insecure direct object reference (IDOR).
## Remediation
A fix was pushed into the \`master\` branch but not yet published.
## References
- [GitHub Commit](https://github.com/idno/known/pull/3101/commits/0b06c2fb05dcb9aa154ba078b560021fc07e7c03)
- [GitHub PR](https://github.com/idno/known/pull/3101)
- [Vulnerability Report](https://blog.jitendrapatro.me/multiple-vulnerabilities-in-idno-known-php-cms-software/#The-Vulnerability)
`}</ReactMarkdown>
```

**Reply by rohitcoder** [reply]
Thank you, the solution worked as expected. I realized that I had added multiple blank spaces in front of each markdown line, which was also causing HTML not to render and be interpreted as code.

**Comment by ChristianMurphy** [comment]
Welcome @rohitcoder! :wave: 
Sorry you ran into a spot of trouble.
Could you share a bit more about the build tools and versions of `react-markdown` and `remark-gfm` you are using?

I tried replicating your issue with Create React App and `react-markdown` version 8.0.5.
And it renders as expected https://codesandbox.io/s/epic-river-wpt6gq

For additional ideas consider creating a minimal reproducible example https://stackoverflow.com/help/minimal-reproducible-example
In stackblitz or codesandbox, some starters: https://codesandbox.io/s/github/remarkjs/.github/tree/main/sandbox-templates/react-markdown-with-create-react-app and https://stackblitz.com/github/remarkjs/.github/tree/main/sandbox-templates/react-markdown-with-create-react-app?file=package.json

**Reply by rohitcoder** [reply]
I am using Next.JS, thanks for your response 🙏🏻, my problem got resolved in above thread :)
